### PR TITLE
Support dot and slash separators in TTree:GetLeaf()

### PR DESCRIPTION
**Pull request by amadio**
See [ROOT-9254](https://sft.its.cern.ch/jira/browse/ROOT-9254) for more information.

### CODE REVIEW

**Comment by amadio** [comment]
I see on my side that some tests are failing. I will investigate what is the problem...

File: tree/tree/src/TTree.cxx [hunk 1]
```diff
@@ -5961,31 +5961,35 @@ TLeaf* TTree::GetLeaf(const char* branchname, const char *leafname)
...
-   }
+   std::string path(name);
+
+   const auto sep = path.find_last_of("/.");
+
+   if (sep == std::string::npos)
+      return GetLeafImpl(nullptr, name);
+
+   auto branch = path.substr(0, sep);
+   auto leaf   = path.substr(sep + 1);
```
Comment (pcanal) [inline tree/tree/src/TTree.cxx]: Is this temporary/copy necessary or can we still use ```path.c_str()+ sep +1``` (or something similar)?

Comment (amadio) [reply tree/tree/src/TTree.cxx]: No, we can use `name+sep+1` if you'd rather have that.

**Comment by amadio** [comment]
This is not needed, functionality has been implemented elsewhere for TDataFrame.

**Comment by bluehood** [comment]
@amadio @pcanal @dpiparo does this mean that we don't care about https://sft.its.cern.ch/jira/projects/ROOT/issues/ROOT-9254 anymore?

**Comment by amadio** [comment]
@bluehood No, only that the functionality in this branch is not needed anymore for that, but other changes will be needed and I am working on those. I will create a new pull request later with my changes.

**Comment by bluehood** [comment]
@amadio I don't understand, as far as I can tell this PR was solving _exactly_ ROOT-9254

**Comment by amadio** [comment]
@bluehood Yes, that was the plan, but I think that it actually breaks things as it was done here. I will look into another way to solve the problem.

**Comment by pcanal** [comment]
@amadio for the record, what was broken by the PR? (i.e. was it inadvertent or fundamental?)

**Comment by amadio** [comment]
@pcanal, Something broke in roottest (things that don't run in the PR), but I don't remember the test names.

**Comment by pcanal** [comment]
Wouldn't it help in finding a better/alternative solution to understand the nature/mechanics/reason of the failure?

**Comment by amadio** [comment]
The problem is not in `GetLeaf()`, so we do not need to add support for this there, especially since it breaks tests. We just have to do the right name split in TDF and pass that to another overload of `GetLeaf()` that takes the already split arguments. The real problem is that once branch names with dots start being accepted by TDF, the JIT doesn't work because the string for jitting contains the branch names as is, so things like `double p.x;` show up in the code to be jitted, and that cannot be compiled, so that's what really needs fixing. ROOT-9254 is not needed and could probably just be closed.

**Pull request closed without merging**